### PR TITLE
fix(UI): render Rosetta as block

### LINKED ISSUE #46006: Missing course collapse/expand button for Rosetta Code
### Describe the Issue

Rosetta Code is missing expand/collapse button. 

Issue is caused as a combination of https://github.com/freeCodeCamp/freeCodeCamp/blob/64cb47e1348d63e38a60d78646c1234f1257ecfd/client/src/templates/Introduction/components/block.tsx#L124-L126 with rosetta code having order `3` and challengeType `5`. As such it is treated as projects block.

One possible way of dealing is changing the check above, to include superBlock check. Another might be using different order in meta for Rosetta.

### Affected Page

https://www.freecodecamp.org/learn/coding-interview-prep/

### Steps to Reproduce

1. Go to https://www.freecodecamp.org/learn/coding-interview-prep/
2. Scroll to Rosetta Code

### Expected behavior

Have expand/collapse button for Rosetta Code

### Screenshots

image

### System

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by JordanMooree**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46006

I've added the `isStepBased` property set to true on all step-based projects (in-production) in their `meta.json` files.

Result of Rosetta:
image

### CODE REVIEW

File: client/src/templates/Introduction/components/block.tsx [hunk 1]
```diff
@@ -120,10 +120,7 @@ export class Block extends Component<BlockProps> {
     });
 
     const isProjectBlock = challenges.some(({ challenge }) => {
-      const isJsProject =
-        [3, 6, 10, 14, 17].includes(challenge.order) &&
-        challenge.challengeType === 5;
-
+      const isJsProject = challenge.isStepBased === true;
```
Comment (ieahleen) [inline client/src/templates/Introduction/components/block.tsx:123]:
```suggestion
      const isJsProject = challenge.isStepBased;
```
feels redundant

**Comment by JordanMooree** [comment]
#45522 has a bit of relation to the requests made [in response to this pull request](https://github.com/freeCodeCamp/freeCodeCamp/pull/46077#issuecomment-1133285058)

**Comment by JordanMooree** [comment]
I've updated all `meta.json` files with the `isStepBased` property. I think it's ready for review @ojeytonwilliams @ShaunSHamilton

**Comment by ojeytonwilliams** [comment]
Hey @JordanMooree I misunderstood what was going on here and gave you some bad advice. Sorry about that!

There are two issues here.  Firstly what you addressed by adding `isStepBased`, namely that the client is performing ad-hoc logic. Secondly, that there isn't a good way to determine which challenges are JS projects.

With those two issues in mind, I think the way to go is as follows.  We have a challenge type `bonfire` that is functionally equivalent to the challenge type `js`.  I'd rename that `jsProject`, convert all the *other* challenges that have challenge type 5 (currently `bonfire`) to challenge type 1 (`js`) and then we can expand the following logic (from https://github.com/freeCodeCamp/freeCodeCamp/pull/46087)

https://github.com/freeCodeCamp/freeCodeCamp/blob/469bac51a6683c22195aae10b93b17c2ec086aa3/utils/challenge-types.js#L44-L54

to include `jsProject`.  Then `isProject` can be used in the client and should be reliable.  With this in place we'd no-longer need to add `isStepBased`.

Let me know if any of that is unclear and, again, sorry for leading you in the wrong direction.

**Pull request closed without merging**